### PR TITLE
feat(segment): analytics plugin for Segment

### PR DESCRIPTION
**Pull request by gorkem**
Initial work . Set up the analytics plugin cleanup and update from the previous work.

### CODE REVIEW

**Review by schultzp2020 — CHANGES_REQUESTED** [review]
Don't forget to run `yarn prettier --write ./plugins/analytics-provider-segment/`

File: plugins/analytics-provider-segment/src/apis/implementations/AnalyticsApi/Segment.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    this.analytics = new AnalyticsBrowser();
+    this.analytics.load({ writeKey: writeKey });
+
+  }
+
+  /**
+   * Instantiate a fully configured Segment API implementation.
+   */
+  static fromConfig(config: Config, identityApi?: IdentityApi) {
+    const writeKey = config.getString('app.analytics.segment.writeKey');
```
[the remark this answers is not in the document]
Comment (gorkem) [reply plugins/analytics-provider-segment/src/apis/implementations/AnalyticsApi/Segment.ts:47]: Makes sense, this is the part I did not touch from the old PR

**Review by schultzp2020 — APPROVED** [review]
LGTM

File: plugins/analytics-provider-segment/package.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
+{
+  "name": "@janus-idp/plugin-analytics-provider-segment",
```
Comment (tumido) [inline plugins/analytics-provider-segment/package.json:2]: Please change to maintain naming convention

```suggestion
  "name": "@janus-idp/backstage-plugin-analytics-provider-segment",
```

or 

```suggestion
  "name": "@janus-idp/backstage-plugin-segment",
```

File: plugins/analytics-provider-segment/package.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  "version": "0.1.0",
+  "main": "src/index.ts",
+  "types": "src/index.ts",
+  "license": "Apache-2.0",
+  "private": true,
```
Comment (tumido) [inline plugins/analytics-provider-segment/package.json:7]: Please remove

```suggestion
```

File: plugins/analytics-provider-segment/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright 2022 The Backstage Authors
```
Comment (tumido) [inline plugins/analytics-provider-segment/src/index.ts:2]: I think this should read in all files

```suggestion
 * Copyright 2023 The Janus IDP Authors
```

**Review by tumido — COMMENTED** [review]
I have just a few formal suggestions, see inline. Otherwise LGTM

File: plugins/analytics-provider-segment/package.json (hunk 2 above)
Comment (gorkem) [reply plugins/analytics-provider-segment/package.json:2]: I think the original author just followed the convention that was used with the Google Analytics plugin upstream. I am OK with the change but we need to update the README etc too

File: plugins/analytics-provider-segment/src/index.ts (hunk 4 above)
Comment (gorkem) [reply plugins/analytics-provider-segment/src/index.ts:2]: We are picking up some of the code from a closed PR upstream. I am not sure if we should change the original copyright rather add Janus

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**